Docs/Encoding.md

```diff
@@ -0,0 +1,193 @@
+| -------------- |  ------------- | -------------- | ------------------ |
+| 1              |  L + 1         | `rrr1`         | `.g` <sup>1</sup>  |
+| 2              |  L + A         | `rrrg`         | `.ga` <sup>1</sup> |
+| 3              |  RGB + 1       | `rgb1`         | `.rgb`             |
+| 4              |  RGB + A       | `rgba`         | `.rgba`            |
+
+**1:** Sampling from `g` is preferred to sampling from `r` because it
+allows a single shader to be compatible with ASTC, BC1, or ETC formats. BC1 and
+ETC1 store endpoints as RGB 565, so the `g` channel will have higher precision.
+For ASTC it doesn't actually make any difference; the same single channel
+luminance will be returned for all three of the `.rgb` channels.
+
+## Equivalence with other formats
+
+Based on these channel encoding requirements we can now derive the the ASTC
+coding equivalents for most of the other texture compression formats in common
+use today.
+
+| Formant  | ASTC Coding Swizzle  | ASTC Sampling Swizzle | Notes            |
+| -------- | -------------------- | --------------------- | ---------------- |
+| BC1      | `rgba` <sup>1</sup>  | `.rgba`               |                  |
+| BC3      | `rgba`               | `.rgba`               |                  |
+| BC3nm    | `gggr`               | `.ag`                 |                  |
+| BC4      | `rrr1`               | `.r1`                 |                  |
+| BC5      | `rrrg`               | `.ra` <sup>2</sup>    |                  |
+| BC6      | `rgb1`               | `.rgb`                | HDR profile only |
+| BC7      | `rgba`               | `.rgba`               |                  |
+| EAC_R11  | `rrr1`               | `.r`                  |                  |
+| EAC_RG11 | `rrrg`               | `.ra` <sup>2</sup>    |                  |
+| ETC1     | `rgba`               | `.rgba`               |                  |
+| ETC2     | `rgba` <sup>1</sup>  | `.rgb`                |                  |
+| ETC2+EAC | `rgba`               | `.rgba`               |                  |
+| ETC2+EAC | `rgba`               | `.rgba`               |                  |
+
+**1:** ASTC has no direct equivalent of the 1-bit punch-through alpha encoding
+supported by BC1 or ETC2; if alpha is present it will be a full alpha channel.
+
+**2:** ASTC relies on using the L+A color endpoint type for coding efficiency
+for two channel data. It therefore has no direct equivalent of a two-plane
+format sampled though the `.rg` channels such as BC5 or EAC_RG11. This can
+be emulated by setting texture channel swizzles in the runtime API - e.g. via
+`glTexParameteri()` for OpenGL ES - although it has been noted that API
+controlled swizzles are not available in WebGL.
+
+# Other Considerations
+
+This section outlines some of the other things to consider when encoding
+textures using ASTC.
+
+## Encoding non-correlated channels
+
+ASTC can optionally assign any single color channel to a separate
```

Comment:
Revise sentence

Reply:
Can you clarify the problem here? It reads OK to me, but that's possibly just because I wrote it ...

Thanks!

Reply:
Original text:

> ASTC can optionally assign any single color channel to a separate non-correlated partition to the other three channels, chosen on a block-by-block basis. In the general case there is no need for non-correlated data to be in a specific channel in the data set; the compressor will assign the non-correlated channel on demand if it is needed.

Attempted reword:

> Most other texture compression formats have a static channel assignment in terms of the expected data correlation. For example, ETC2+EAC assumes that RGB are always correlated and that alpha is non-correlated. ASTC can automatically encode data as either fully correlated across all 4 channels, or with any one channel assigned to a separate non correlated partition to the other three. The non-corellated channel can be changed on a block-by-block basis, so the compressor can dynamically adjust the coding based on the data present in the image. This means that there is no need for non-correlated data to be stored in a specific channel in the input image.

Reply:
It was this part "to a separate non-correlated partition to the other three channels" - the reword looks much better.